ac.rkt

```diff
@@ -131,8 +162,17 @@
          (ac (list 'no (cons (cadar s) (cdr s))) env)]
         [(eq? (xcar (xcar s)) 'andf) (ac-andf s env)]
         [(pair? s) (ac-call (car s) (cdr s) env)]
+        [(syntax? s) s]
         [#t (err "Bad object in expression" s)]))
 
+(defarc (ac stx (env (env*)) (ns main-namespace))
+  (parameterize ((env* env))
+    (let* ((s (syn stx))
+           (e (syntax->datum s))
+           (expr (ac* s e env)))
+      (parameterize ((current-namespace ns))
+        (namespace-syntax-introduce (syn expr stx))))))
+
```

Comment:
Gosh, this is where the magic happens, isn't it? A well-placed `namespace-syntax-introduce`.

I just noticed you call `(ac* s e env)` here, but `ac*` is defined by `(defarc (ac* e s env) ...)`.